### ISSUE TITLE
Missing `puppeteer-extra` peer dependency

### ISSUE THREAD

**Issue by rtritto** [berstend/puppeteer-extra#645]
Transitive `puppeteer-extra` peer dependency must be included in each dependency:
- `puppeteer-extra-plugin-anonymize-ua`
- `puppeteer-extra-plugin-block-resources`
- `puppeteer-extra-plugin-click-and-wait`
- `puppeteer-extra-plugin-devtools`
- `puppeteer-extra-plugin-flash`
- `puppeteer-extra-plugin-font-size`
- `puppeteer-extra-plugin-repl`
- `puppeteer-extra-plugin-stealth`
- `puppeteer-extra-plugin-user-data-dir`
- `puppeteer-extra-plugin-user-preferences`

## Changes
`package.json`
```diff
"peerDependencies": {
+  "puppeteer-extra": "*",
  ...
}
```

**Comment by berstend** [comment]
Is this a hypothetical issue or a real-life one? npm changed their behavior around peer dependencies a while back, if I recall correctly

**Comment by berstend** [comment]
Given we now introduced `playwright-extra` (#664) which is able to load in those plugins we probably want to do the exact opposite and remove all peer dependencies 😄 

Therefore closing this for now

**Comment by rtritto** [comment]
@berstend I don't know if you should consider to use optional peer dependency:
```json
"peerDependencies": {
  "puppeteer-extra": "*",
},
"peerDependenciesMeta": {
  "puppeteer-extra": {
    "optional": true
  }
}
```

**Comment by berstend** [comment]
The whole situation with peer dependencies is messy :-/

https://github.com/npm/cli/issues/1247#issuecomment-688360885

I wonder what the best behavior from our perspective would be, warning but not auto-installing?

**Comment by rtritto** [comment]
With _optional = true_, peer dependencies will not be installed. Source: https://github.com/npm/rfcs/issues/221#issuecomment-695127530

We can ask some help to arcanis and merceyz from [yarn](https://github.com/yarnpkg/berry).

**Comment by berstend** [comment]
Fixed in `puppeteer-extra@3.3.2`, apologies 😄
